lib/search-field/index.jsx

```diff
@@ -36,58 +38,68 @@ export class SearchField extends Component {
     }
   }
 
+  doSearch = query => {
+    this.setState({ query, searchSelected: true });
+    this.props.onSearch(query);
+  };
+
   interceptEsc = event => {
     if (KEY_ESC === event.keyCode) {
       if (this.state.query === '') {
         this.inputField.blur();
       }
       this.clearQuery();
     }
+    if (KEY_ENTER === event.keyCode) {
+      this.doSearch(this.state.query);
+    }
   };
 
   storeInput = r => (this.inputField = r);
 
-  debouncedSearch = debounce(query => this.props.onSearch(query), SEARCH_DELAY);
-
   update = ({ target: { value: query } }) => {
-    this.setState({ query });
-    this.debouncedSearch(query);
+    this.setState({ query, searchSelected: false });
   };
 
   clearQuery = () => {
-    this.setState({ query: '' });
-    this.debouncedSearch('');
-    this.debouncedSearch.flush();
+    this.setState({ query: '', searchSelected: false });
+    this.props.onSearch('');
   };
 
   render() {
     const { isTagSelected, placeholder } = this.props;
-    const { query } = this.state;
+    const { query, searchSelected } = this.state;
     const hasQuery = query && query.length > 0;
+    const shouldShowSuggestions = hasQuery && !searchSelected;
 
     const screenReaderLabel =
       'Search ' + (isTagSelected ? 'notes with tag ' : '') + placeholder;
 
     return (
-      <div className="search-field">
-        <input
-          aria-label={screenReaderLabel}
-          ref={this.storeInput}
-          type="text"
-          placeholder={placeholder}
-          onChange={this.update}
-          onKeyUp={this.interceptEsc}
-          value={query}
-          spellCheck={false}
-        />
-        <button
-          aria-label="Clear search"
-          hidden={!hasQuery}
-          onClick={this.clearQuery}
-        >
-          <SmallCrossIcon />
-        </button>
-      </div>
+      <Fragment>
+        <div className="search-field">
+          <input
+            aria-label={screenReaderLabel}
+            ref={this.storeInput}
+            type="text"
+            placeholder={placeholder}
+            onChange={this.update}
+            onKeyUp={this.interceptEsc}
+            value={query}
+            spellCheck={false}
+          />
+          <button
+            aria-label="Clear search"
+            hidden={!hasQuery}
+            onClick={this.clearQuery}
+          >
+            <SmallCrossIcon />
+          </button>
+          {shouldShowSuggestions && (
+            <SearchSuggestions query={query} onSearch={this.doSearch} />
```

Comment:
Likewise, if we really want to rely on Redux state for providing this information we won't have to pass it along here. Thankfully this doesn't imply any changes to `SearchSuggestions` other than we add the appropriate props to `connect()` instead of passing them _through_ components like this.

Reply:
[Ditto](https://github.com/Automattic/simplenote-electron/pull/1643#discussion_r336329225).